guidelines.md

```diff
@@ -0,0 +1,91 @@
+
+```$ git clone https://github.com/<your-account-username>/<your-forked-project>.git```
+Now you have the project on your local machine.
+
+### ADD A REMOTE (UPSTREAM) TO ORIGINAL PROJECT REPOSITORY 
+***
+Remote means the remote location of project on Github. By cloning, we have a remote called origin which points to your forked repository. Now we will add a remote to the original repository from where we had forked.
+
+```$ cd <your-forked-project-folder>```
+```$ git remote add upstream https://github.com/<author-account-username>/<project>.git```
+You will see the benefits of adding remote later.
+
+### SYNCHRONIZING YOUR FORK -
+***
+Open Source projects have a number of contributors who can push code anytime. So it is necessary to make your forked copy equal with the original repository. The remote added above called Upstream helps in this.
+
+```$ git checkout master```
+```$ git fetch upstream```
+```$ git merge upstream/master```
+```$ git push origin master```
+The last command pushes the latest code to your forked repository on Github. The origin is the remote pointing to your forked repository on github.
+
+### CREATE A NEW BRANCH FOR A FEATURE OR BUGFIX -
+***
+Normally, all repositories have a master branch which is considered to remain stable and all new features should be made in a separate branch and after completion merged into master branch. So we should create a new branch for our feature or bugfix and start working on the issue.
+
+```$ git checkout -b <feature-branch>```
+This will create a new branch out of master branch. Now start working on the problem and commit your changes.
+
+```$ git add --all```
+```$ git commit -m "<commit message>"```
+The first command adds all the files or you can add specific files by removing -a and adding the file names. The second command gives a message to your changes so you can know in future what changes this commit makes. If you are solving an issue on original repository, you should add the issue number like #35 to your commit message. This will show the reference to commits in the issue.
+
+### REBASE YOUR FEATURE BRANCH WITH UPSTREAM-
+***
+It can happen that your feature takes time to complete and other contributors are constantly pushing code. After completing the feature your feature branch should be rebase on latest changes to upstream master branch.
+
+```$ git checkout <feature-branch>```
+```$ git pull --rebase upstream master```
+Now you get the latest commits from other contributors and check that your commits are compatible with the new commits. If there are any conflicts solve them.
+
+### SQUASHING YOUR COMMITS-
+***
+You have completed the feature, but you have made a number of commits which make less sense. You should squash your commits to make good commits.
+
+```$ git rebase -i HEAD~5```
+This will open an editor which will allow you to squash the commits.
```

Comment:
This sentence continuous in the same line as the prev git command (when rendering). Start this in a new line.
This issue persists in many lines that come after a command. I have only mentioned a few in my comments.